src/connection.js

```diff
@@ -585,8 +594,12 @@ class Connection extends EventEmitter {
 
   requestTimeout() {
     this.requestTimer = undefined;
-    this.messageIo.sendMessage(TYPE.ATTENTION);
-    return this.transitionTo(this.STATE.SENT_ATTENTION);
+    this.cancel();
+  }
+
+  cancelTimeout() {
+    this.cancelTimer = undefined;
+    return this.close();
```

Comment:
I don't think the `return` here is necessary.

Also, on top of that, I'm not sure whether just calling `this.close` is the best course of action here. The in-flight request that did not get cancelled in time will get a `RequestError`, but it will not be clear that this error was passed because cancellation timed out.